File: python/circuit.py
#!/usr/bin/python

from constants import *
from Cell import Cell
from Wire import Wire

wires = []

for i in range(numCols):
    for j in range(numRows):
        cells.append(Cell(i,j))

numTries = 100
for i in range(numWires):
    tries = 0
    start = cells[random.randint(0,len(cells)-1)]
    while start.open is False:
        tries += 1
        if tries > numTries:
            break
        start = cells[random.randint(0,len(cells)-1)]
    wire = Wire(start)
    wire.create_wire()
    if len(wire.cells) > minLineLength:
        wires.append(wire)

def draw_grid():
    ctx.set_source_rgba(0,0,0,0.2)
    ctx.set_line_width(2)
    for i in range(numCols):
        ctx.move_to(i * cellSize,0)
        ctx.line_to(i * cellSize,numRows*cellSize)
        ctx.stroke()
    for j in range(numRows):
        ctx.move_to(0,j * cellSize)
        ctx.line_to(numCols*cellSize,j * cellSize)
        ctx.stroke()

ctx.rectangle(0,0,W,H)
ctx.set_source_rgb(BOARD_COL[0], BOARD_COL[1], BOARD_COL[2])
ctx.fill()
for wire in wires:
    wire.draw_wire()
    for fork in wire.forks:
        if fork.start != wire.cells[len(wire.cells) - 1]:
            fork.create_wire()
            if len(fork.cells) > 1:
                fork.draw_wire()
#draw_grid()
surface.write_to_png('./circuit.png')

File: p5js/circuit.js
let directions = [
  [1,0],[1,1],
  [0,1],[-1,1],
  [-1,0],[-1,-1],
  [0,-1],[1,-1]
]

let BOARD_COL = [25,75,25]
let WIRE_COL = [150,128,50]
let numRows = 24
let numCols = 36
let cellSize = 20
let cells = []
let wires = []

let MaxLineLength = 80;
let numWires = 80;
let lineWidth = cellSize/4;
let W = numCols * cellSize;
let H = numRows * cellSize

for(let i = 0; i < numCols; i += 1) {
  for(let j = 0; j < numRows; j += 1) {
    cells.push(new cell(i,j));
  }
}

for(let i = 0; i < numWires; i += 1) {
  var w = new Wire(cells[Math.floor(Math.random() * cells.length)]);
  w.create_wire();
  wires.push(w);
}

function setup() {
  createCanvas(W,H);
  background(BOARD_COL);
}

function draw() {
  background(BOARD_COL);
  //draw_grid();
  fill(255);
  for(let i = 0; i < wires.length; i += 1) {
    wires[i].draw_wire();
    wires[i].draw_vias();
  }
}

function get_cell(x,y) {
  for(let i = 0; i < cells.length; i += 1) {
    if(cells[i].x == x && cells[i].y == y) {
      return cells[i];
    }
  }
}

function draw_grid() {
  for(let i = 0; i < numCols; i += 1) {
    line(i*cellSize,0,i*cellSize,numRows*cellSize);
    stroke(255,64);
  }

  for(let j = 0; j < numRows; j += 1) {
    line(0,j*cellSize,numCols*cellSize,j*cellSize);
    stroke(255,64);
  }
}

Number.prototype.mod = function(n) {
      return ((this%n)+n)%n;
};
